### PR TITLE
Update Azure SQL server guide with Entra ID

### PR DESCRIPTION
**Pull request by greedy52**

### CODE REVIEW

File: docs/pages/enroll-resources/database-access/enroll-azure-databases/azure-sql-server-ad.mdx [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+Azure identity with permissions to retrieve authentication tokens from
+Microsoft Entra ID. When a user connects to SQL Server with Teleport, the Teleport
 Database service authenticates with Azure AD, then uses an authentication token
 to connect to SQL Server. The Database Service then forwards user traffic to the
 database.
 
 <Tabs>
 <TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
-![Teleport Database Access Azure SQL Server Azure Active Directory Self-Hosted](../../../../img/database-access/guides/sqlserver/sql-aad.png)
+![Teleport Database Access Azure SQL Server Microsoft Entra Self-Hosted](../../../../img/database-access/guides/sqlserver/sql-aad.png)
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/enroll-azure-databases/azure-sql-server-ad.mdx:19]:
```suggestion
![Access Azure SQL Server Microsoft Entra Self-Hosted](../../../../img/database-access/guides/sqlserver/sql-aad.png)
```
To appease the linter

File: docs/pages/enroll-resources/database-access/enroll-azure-databases/azure-sql-server-ad.mdx [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
 </TabItem>
 <TabItem scope={["cloud"]} label="Teleport Enterprise Cloud">
-![Teleport Database Access Azure SQL Server Azure Active Directory Cloud](../../../../img/database-access/guides/sqlserver/cloud-sql-aad.png)
+![Teleport Database Access Azure SQL Server Microsoft Entra Cloud](../../../../img/database-access/guides/sqlserver/cloud-sql-aad.png)
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/enroll-azure-databases/azure-sql-server-ad.mdx:22]:
```suggestion
![Access Azure SQL Server Microsoft Entra Cloud](../../../../img/database-access/guides/sqlserver/cloud-sql-aad.png)
```
To appease the linter

**Review by ptgott — APPROVED** [review]

File: docs/img/database-access/guides/sqlserver/cloud-sql-aad.png
Comment (mmcallister) [inline docs/img/database-access/guides/sqlserver/cloud-sql-aad.png:1]: These are using the old Teleport logo - you can replace it with this one - 
new-logo-solid-square-white-border

Comment (greedy52) [reply docs/img/database-access/guides/sqlserver/cloud-sql-aad.png:1]: good catch! updated now!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**